### PR TITLE
Encapsulate Worker.batched_stream.send()

### PR DESCRIPTION
**Pull request by crusaderky**
- Part of #5736 
- Blocked by #6474

Encapsulate calls to `self.batched_stream.send` in a way that
1. tweaks and harmonizes the many repeated tests for running comms
2. will be easier to abstract with the WorkerBase class (see #5736)

### CODE REVIEW

File: distributed/worker.py [hunk 1]
```diff
@@ -984,20 +982,13 @@ def status(self, value):
...
-        ):
-            self.batched_stream.send(
-                {
-                    "op": "worker-status-change",
-                    "status": self._status.name,
-                    "stimulus_id": stimulus_id,
-                },
-            )
-        elif self._status != Status.closed:
-            self.loop.call_later(0.05, self._send_worker_status_change, stimulus_id)
```
Comment (crusaderky) [inline distributed/worker.py:1000]: This is a race condition waiting to happen

File: distributed/worker.py [hunk 2]
```diff
@@ -1100,6 +1091,22 @@ def _to_dict(self, *, exclude: Container[str] = ()) -> dict:
...
+            msgpack-serializable message to send to the scheduler.
+            Must have a 'op' key which is registered in Scheduler.stream_handlers.
+        """
+        if (
+            self.batched_stream
+            and self.batched_stream.comm
+            and not self.batched_stream.comm.closed()
+        ):
+            self.batched_stream.send(msg)
+
```
Comment (crusaderky) [inline distributed/worker.py:1109]: This method will become an `@abc.abstractmethod` in the WorkerBase class

File: distributed/worker.py [hunk 3]
```diff
@@ -2840,13 +2842,7 @@ def transitions(self, recommendations: Recs, *, stimulus_id: str) -> None:
             for ts in tasks:
                 self.validate_task(ts)
 
-        if self.batched_stream.closed():
-            logger.debug(
-                "BatchedSend closed while transitioning tasks. %d tasks not sent.",
-                len(instructions),
-            )
-        else:
-            self._handle_instructions(instructions)
```
Comment (crusaderky) [inline distributed/worker.py:2849]: This is old cruft from when the only possible instructions were messages

**Comment by crusaderky** [comment]
Failure in test_transition_counter_max_worker is fixed by https://github.com/dask/distributed/pull/6474

**Comment by mrocklin** [comment]
cc @gjoseph92 for review if you have time

File: distributed/worker.py (hunk 1 above)
Comment (gjoseph92) [inline distributed/worker.py:1000]: This retry will now be lost. I imagine this case occurs if the worker's status changes before it's connected to the scheduler. I'm not sure if this is important.

As part of https://github.com/dask/distributed/issues/6389, we could pretty easily make `BatchedSend` support queuing messages before it's been started.

File: distributed/worker.py [hunk 4]
```diff
@@ -1100,6 +1091,22 @@ def _to_dict(self, *, exclude: Container[str] = ()) -> dict:
     # External Services #
     #####################
 
+    def batched_send(self, msg: dict[str, Any]) -> None:
+        """Send a fire-and-forget message to the scheduler through bulk comms.
```
Comment (gjoseph92) [inline distributed/worker.py:1095]:
```suggestion
        """Send a fire-and-forget message to the scheduler through bulk comms.
        
        If we're not currently connected to the scheduler, the message will be silently dropped!
```
Just seems important to make clear.

File: distributed/worker.py (hunk 1 above)
Comment (crusaderky) [reply distributed/worker.py:1000]: This used to be necessary when I implemented it - the WorkerState on the scheduler side would never emerge out of initialising state. Now it is not needed anymore.

**Review by gjoseph92 — APPROVED** [review]

**Pull request merged**